lib/widgets/container.dart

```diff
@@ -13,15 +24,29 @@ import '../args/edge_insets.dart';
 extension ContainerConverter on Container {
   /// Converts the [Container] to a [pw.Container].
   Future<pw.Container> toPdfWidget(pw.Widget? child) async => pw.Container(
-        alignment: (alignment as Alignment?)?.toPdfAlignment(),
+        alignment: _convertAlignment(alignment),
         decoration: await (decoration as BoxDecoration?)?.toPdfBoxDecoration(),
         color: color?.toPdfColor(),
         constraints: constraints?.toPdfBoxConstraints(),
         foregroundDecoration: await (foregroundDecoration as BoxDecoration?)
             ?.toPdfBoxDecoration(),
-        margin: (margin as EdgeInsets?)?.toPdfEdgeInsets(),
-        padding: (padding as EdgeInsets?)?.toPdfEdgeInsets(),
+        margin: _convertEdge(margin),
+        padding: _convertEdge(padding),
         transform: transform,
         child: child,
       );
+
+  _convertAlignment(AlignmentGeometry? alignment) {
+    if (alignment is Alignment) return alignment.toPdfAlignment();
+    if (alignment is AlignmentDirectional) return alignment.toPdfAlignment();
+    debugPrint('Unsupported AlignmentGeometry: $this; defaulting to null');
+    return null;
+  }
+
+  _convertEdge(EdgeInsetsGeometry? edge) {
+    if (edge is EdgeInsetsDirectional) return edge.toPdfEdgeInsetsDirectional();
+    if (edge is EdgeInsets) return edge.toPdfEdgeInsets();
+    debugPrint('Unsupported EdgeInsetsGeometry: $this; defaulting to null');
+    return null;
+  }
```

Comment:
`_convertAlignment` & `_convertEdge` will be obsolete and replaced by the `AlignmentGeometryConverter`s `toPdfAlignment` and the `EdgeInsetsGeometryConverter`s `toPdfEdgeInsets`.

Reply:
Suggestion implemented!